### PR TITLE
[kube-state-metrics] add service load balancer IP ranges

### PR DESCRIPTION
**Pull request by obrienrobert**
#### What this PR does / why we need it
Add `loadBalancerSourceRanges` to `kube-state-metrics` Helm chart. Inspired by [pull/2502](https://github.com/prometheus-community/helm-charts/pull/2502).

#### Which issue this PR fixes

*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*

- fixes #

#### Checklist

- [ ] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [ ] Chart Version bumped
- [ ] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Review by dotdc — CHANGES_REQUESTED** [review]
New features should raise minor version, could you update to 4.23.0?

**Comment by obrienrobert** [comment]
@dotdc good catch, I should have known. Updated now.

**Review by dotdc — DISMISSED** [review]
LGTM

**Comment by dotdc** [comment]
Please fix the DCO @obrienrobert 
https://github.com/prometheus-community/helm-charts/pull/2672/checks?check_run_id=9415714764

File: charts/kube-state-metrics/values.yaml [hunk 1]
```diff
@@ -29,6 +29,7 @@ service:
   type: ClusterIP
   nodePort: 0
   loadBalancerIP: ""
+  loadBalancerSourceRanges: []  # Only allow access to the loadBalancerIP from these IPs
```
Comment (monotek) [inline charts/kube-state-metrics/values.yaml:32]: please put comment above variable

**Review by monotek — CHANGES_REQUESTED** [review]

**Review by monotek — APPROVED** [review]

**Review by dotdc — APPROVED** [review]
LGTM

**Comment by dotdc** [comment]
Thank you for your contribution @obrienrobert !

**Pull request merged**